### PR TITLE
Desktop layout

### PR DESCRIPTION
**Pull request by b-avb**
- Rooms list be side to the message timeline.
- Clamp the width of components in views register/login and profile/verify
- Threads go on a side panel besides the main conversation.
- Remove unnecessary inline styles

### CODE REVIEW

File: src/pages/profile/profile.rs [hunk 1]
```diff
@@ -192,208 +182,177 @@ pub fn Profile(cx: Scope) -> Element {
...
-                                                    info!("{x:?}");
+                                            if let Some(y) = attach.get() {
+                                                let x = client.get().account().upload_avatar(&mime::IMAGE_PNG, &y.data).await;
+                                                info!("{x:?}");
+                                                match x {
+                                                    Ok(url)=> {
+                                                        let x = client.get().account().set_avatar_url(Some(&url)).await;
+                                                        info!("{x:?}");
+                                                    }
+                                                    Err(_)=> {}
```
Comment (olanod) [inline src/pages/profile/profile.rs:267]: Same thing with the error ... Rust complains about unused results for a reason, ignoring errors is a recipe for disaster

File: src/pages/profile/profile.rs [hunk 2]
```diff
@@ -192,208 +182,177 @@ pub fn Profile(cx: Scope) -> Element {
...
+                        p {
+                            class: "profile__content",
+                            "{i18n_get_key_value(&i18n_map, key_management_deactivate_label)}"
+                        }
+                        div {
+                            class: "profile__cta",
+                            Button {
+                                text: "{i18n_get_key_value(&i18n_map, key_management_deactivate_cta_deactivate)}",
+                                on_click: move |_| {
+                                    // cx.spawn({
```
Comment (olanod) [inline src/pages/profile/profile.rs:343]: ✂️ ?

File: src/pages/signup.rs [hunk 3]
```diff
@@ -270,151 +270,153 @@ pub fn Signup(cx: Scope) -> Element {
...
+        div {
+            class: "page--clamp",
+            if auth.get().data.server.is_none() {
+                rsx!(LoginForm {
+                    title: "{i18n_get_key_value(&i18n_map, key_signup_chat_homeserver_message)}",
+                    description: "{i18n_get_key_value(&i18n_map, key_signup_chat_homeserver_description)}",
+                    button_text: "{i18n_get_key_value(&i18n_map, key_signup_chat_homeserver_cta)}",
+                    emoji: "🛰️",
+                    error: if error.get().is_some() {
+                        error.get().as_ref()
```
Comment (olanod) [inline src/pages/signup.rs:283]: I suspect the if block is unnecessary, isn't `.as_ref()` returning an option?

File: src/pages/signup.rs [hunk 4]
```diff
@@ -270,151 +270,153 @@ pub fn Signup(cx: Scope) -> Element {
...
+                    ))
+                })
+            } else if auth.get().data.username.is_none() || auth.get().data.password.is_none() {
+                rsx!(LoginForm {
+                    title: "{i18n_get_key_value(&i18n_map, key_signup_chat_credentials_title)}",
+                    description: "{i18n_get_key_value(&i18n_map, key_signup_chat_credentials_description)}",
+                    button_text: "{i18n_get_key_value(&i18n_map, key_signup_chat_credentials_cta)}",
+                    emoji: "✍️",
+                    error: if error.get().is_some() {
+                        error.get().as_ref()
```
Comment (olanod) [inline src/pages/signup.rs:322]: same, unnecessary if block?

File: src/components/organisms/chat/active_room.rs [hunk 5]
```diff
@@ -103,46 +109,129 @@ pub fn ActiveRoom(cx: Scope) -> Element {
...
             }
-        }
 
+            if let Some(t) = timeline_thread.read().deref() {
+                let head_message = &t.thread[t.thread.len() - 1];
+                let x = &head_message.body;
+
+                let title_thread = match x {
+                    TimelineMessageType::Image(file) => {
+                        String::from(file.body.clone())
```
Comment (olanod) [inline src/components/organisms/chat/active_room.rs:169]: looks redundant to have `String::from` and `clone` either one of the should be enough

File: src/components/molecules/list.rs [hunk 6]
```diff
@@ -17,436 +18,346 @@ use crate::{components::{atoms::{
...
-                }
-            )
+    use_effect(cx, (on_scroll,), |(_,)| {
+        let props = cx.props.to_owned();
+
+        if *on_scroll.get() && !cx.props.is_loading {
+            props.on_scroll.call(ListEvent {  });
         }
-        
+        async move {}
```
Comment (olanod) [inline src/components/molecules/list.rs:55]: Is this intentional?

**Review by olanod — CHANGES_REQUESTED** [review]
Nice progress :) Although for future PRs it would be better to have even smaller more self contained changes ;)

File: src/hooks/use_send_message.rs [hunk 7]
```diff
@@ -32,23 +32,31 @@ pub fn use_send_message(cx: &ScopeState) -> &UseSendMessageState {
                     let room_id = RoomId::parse(message_item.room_id).unwrap();
                     let thread_to = threading_to.read().clone();
 
-                    let reply_event_id = if let Some(e) = message_item.reply_to {
-                        Some(EventId::parse(e).unwrap())
-                    } else {
-                        None
+                    let reply_event_id = match message_item.reply_to {
```
[the remark this answers is not in the document]
Comment (b-avb) [reply src/hooks/use_send_message.rs:35]: This was already refactored in other commits that will be presented in another PR. :)

File: src/components/molecules/list.rs (hunk 6 above)
Comment (b-avb) [reply src/components/molecules/list.rs:55]: Yep, effects require something asynchronous to be defined, even if it is empty.

**Review by olanod — APPROVED** [review]

**Pull request merged**